### ISSUE TITLE
NotNan::unchecked_from() could be const

### ISSUE THREAD

**Issue by not-a-seagull** [reem/rust-ordered-float#67]
`NotNan::unchecked_from()` is an unsafe way to bypass the usual `NotNan` mechanism, by directly instantiating the struct with the desired float. However, it is not `const`, despite the fact that it could be, allowing its use in constant contexts.

My use case for this is that I have a `Color` struct:

```
pub struct Color {
    r: NotNan<f32>,
    g: NotNan<f32>,
    b: NotNan<f32>,
    a: NotNan<f32>,
}

impl Color {
    pub unsafe fn new_no_nan_check(r: f32, g: f32, b: f32, a: f32) -> Self {
        Self { r: NotNan::unchecked_new(r), g: NotNan::unchecked_new(g), b: NotNan::unchecked_new(b), a: NotNan::unchecked_new(a) }
    }
}
```

It would be nice if I could make `new_no_nan_check` const, so I could do something similar to:

```
const BLACK: Color = unsafe { Color::new_no_nan_check(0.0, 0.0, 0.0, 1.0) };
const WHITE: Color = unsafe { Color::new_no_nan_check(1.0, 1.0, 1.0, 1.0) };
```

I am willing to implement this feature if there is interest.

**Comment by mbrubeck** [comment]
> I am willing to implement this feature if there is interest.

Yes, that would be welcome!

**Comment by mbrubeck** [comment]
Fixed by #68.
